endpointmanager/pkg/nppesquerier/populatenpiorgs.go

```diff
@@ -0,0 +1,425 @@
+	Healthcare_Provider_Taxonomy_Group_14                                  string
+	Healthcare_Provider_Taxonomy_Group_15                                  string
+}
+
+func ParseNPIdataLine(line []string) NPICsvLine {
+	data := NPICsvLine{
+		NPI:              line[0],
+		Entity_Type_Code: line[1],
+		Provider_Organization_Name_Legal_Business_Name:          line[4],
+		Provider_Other_Organization_Name:                        line[11],
+		Provider_First_Line_Business_Practice_Location_Address:  line[28],
+		Provider_Second_Line_Business_Practice_Location_Address: line[29],
+		Provider_Business_Practice_Location_Address_City_Name:   line[30],
+		Provider_Business_Practice_Location_Address_State_Name:  line[31],
+		Provider_Business_Practice_Location_Address_Postal_Code: line[32],
+		Healthcare_Provider_Taxonomy_Code_1:                     line[47],
+	}
+	return data
+}
+
+func BuildNPIOrgFromNPICsvLine(data NPICsvLine) *endpointmanager.NPIOrganization {
+	npi_org := &endpointmanager.NPIOrganization{
+		NPI_ID:        data.NPI,
+		Name:          data.Provider_Organization_Name_Legal_Business_Name,
+		SecondaryName: data.Provider_Other_Organization_Name,
+		Location: &endpointmanager.Location{
+			Address1: data.Provider_First_Line_Business_Practice_Location_Address,
+			Address2: data.Provider_Second_Line_Business_Practice_Location_Address,
+			City:     data.Provider_Business_Practice_Location_Address_City_Name,
+			State:    data.Provider_Business_Practice_Location_Address_State_Name,
+			ZipCode:  data.Provider_Business_Practice_Location_Address_Postal_Code},
+		Taxonomy: data.Healthcare_Provider_Taxonomy_Code_1}
+	return npi_org
+}
+
+func panicOnErr(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// ReadCsv accepts a file and returns its content as a multi-dimentional type
+// with lines and each column. Only parses to string type.
+func ReadCsv(filename string) ([][]string, error) {
+
+	// Open CSV file
+	f, err := os.Open(filename)
+	if err != nil {
+		return [][]string{}, err
+	}
+	defer f.Close()
+
+	// Read File into a Variable
+	lines, err := csv.NewReader(f).ReadAll()
+	if err != nil {
+		return [][]string{}, err
+	}
+	// return lines without header line
+	return lines[1:], nil
+}
+
+// Parses NPI Org data out of fname, writes it to store and returns the number of organizations processed
+func ParseAndStoreNPIFile(fname string, store *postgresql.Store) int {
+	// Provider organization .csv downloaded from http://download.cms.gov/nppes/NPI_Files.html
+	lines, err := ReadCsv(fname)
+	panicOnErr(err)
+	added_or_updated := 0
+	// Loop through lines & turn into object
+	for _, line := range lines {
+		data := ParseNPIdataLine(line)
+		// We will only parse out organizations (entiy_type_code == 2), not individual providers
+		if data.Entity_Type_Code == "2" {
+			npi_org := BuildNPIOrgFromNPICsvLine(data)
+			err = store.AddOrUpdateNPIOrganization(npi_org)
+			if err != nil {
+				if err != nil {
+					log.Printf("%s", err)
```

Comment:
is there a way you can return this error so that the calling main function can handle it?
if not, provide a logging level.